src/staking/manager.ts

```diff
@@ -0,0 +1,660 @@
+}
+
+interface StakingInputs {
+  finalityProviderPkNoCoordHex: string;
+  stakingAmountSat: number;
+  stakingTimelock: number;
+}
+
+interface InclusionProof {
+  pos: number;
+  merkle: string[];
+  blockHash: string;
+}
+
+export class BabylonBtcStakingManager {
+  private stakingParams: VersionedStakingParams[];
+  private btcProvider: BtcProvider;
+  private network: networks.Network;
+  private bbnProvider: BbnProvider;
+
+  constructor(
+    network: networks.Network,
+    stakingParams: VersionedStakingParams[],
+    btcProvider: BtcProvider,
+    bbnProvider: BbnProvider,
+  ) {
+    this.network = network;
+    this.btcProvider = btcProvider;
+    this.bbnProvider = bbnProvider;
+
+    if (stakingParams.length === 0) {
+      throw new Error("No staking parameters provided");
+    }
+    this.stakingParams = stakingParams;
+  }
+
+  /**
+   * Creates an signed EOI transaction that is ready to be sent to the BBN chain.
+   * @param stakingInput - The staking inputs.
+   * @param feeRate - The fee rate in satoshis per byte.
+   * @returns The signed babylon transaction in base64 format.
+   */
+  async createEoiDelegationBbnTransaction(
+    stakingInput: StakingInputs,
+    feeRate: number,
+  ): Promise<{
+    signedBbnTx: Uint8Array;
+    stakingTx: Transaction;
+  }> {
+    const tipHeight = await this.bbnProvider.getBabylonBtcTipHeight();
+    if (tipHeight === 0) {
+      throw new Error("Babylon BTC tip height cannot be 0");
+    }
+    const inputUTXOs = await this.btcProvider.getUTXOs();
+    if (inputUTXOs.length === 0) {
+      throw new Error("No input UTXOs provided");
+    }
+    const bech32Address = await this.bbnProvider.getBbnAddress();
+    if (!bech32Address) {
+      throw new Error("Babylon address cannot be empty");
+    }
+
+    // Get the param based on the tip height
+    // EOI should always be created based on the BTC tip height from BBN chain
+    const p = getBbnParamByBtcHeight(tipHeight, this.stakingParams);
+    
+    const stakerInfo = await this.btcProvider.getStakerInfo();
+
+    const staking = new Staking(
+      this.network,
+      stakerInfo,
+      p,
+      stakingInput.finalityProviderPkNoCoordHex,
+      stakingInput.stakingTimelock,
+    );
+
+
+      // Create unsigned staking transaction
+      const { transaction } = staking.createStakingTransaction(
+        stakingInput.stakingAmountSat,
+        inputUTXOs,
+        feeRate,
+      );
+
+      // Create delegation message without including inclusion proof
+      const msg = await this.createBtcDelegationMsg(
+        staking,
+        stakingInput,
+        transaction,
+        bech32Address,
+        stakerInfo,
+        p
+      );
+
+      return {
+        signedBbnTx: await this.bbnProvider.signTransaction(msg),
+        stakingTx: transaction,
+      };
+  }
+
+  /**
+   * Creates a signed delegation transaction that is ready to be sent to the 
+   * BBN chain. This is used when a staking transaction is already created and
```

Comment:
```suggestion
   * Babylon chain. This is used when a staking transaction is already created and
```

We don't use bbn anymore. If you want a shorthand, you can use "BABY"